### PR TITLE
fix(cli): don't processed deleted files with `--changed`

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes https://github.com/biomejs/biome/issues/1508

## Test Plan

 I tested locally using our repository with a testing branch

### CODE REVIEW

**Review by arendjr — APPROVED** [review]
Would be nice to have an automated test too.

File: crates/biome_fs/src/fs/os.rs [hunk 1]
```diff
@@ -57,12 +57,27 @@ impl FileSystem for OsFileSystem {
     fn get_changed_files(&self, base: &str) -> io::Result<Vec<String>> {
         let output = Command::new("git")
             .arg("diff")
-            .arg("--name-only")
+            .arg("--name-status")
```
Comment (Conaclos) [inline crates/biome_fs/src/fs/os.rs:60]: We could use `--diff-filter`? For example `git diff --name-only -ddif-filter=d` displays changed files, excluding deleted ones.

Comment (ematipico) [reply crates/biome_fs/src/fs/os.rs:60]: Damn, git is so full of stuff

Comment (ematipico) [reply crates/biome_fs/src/fs/os.rs]: @Conaclos please check again

**Review by Conaclos — APPROVED** [review]
Looks good to me :)

**Pull request merged**